### PR TITLE
refactor(runtime): factor out FsPermissions for fs ops

### PR DESCRIPTION
**Pull request by bartlomieju**
This will help us with moving fs ops to a separate extension crate.

### CODE REVIEW

File: runtime/ops/fs.rs [hunk 1]
```diff
@@ -1826,16 +1933,19 @@ fn op_make_temp_file_sync(
 }
 
 #[op]
-async fn op_make_temp_file_async(
+async fn op_make_temp_file_async<P>(
```
Comment (dsherret) [inline runtime/ops/fs.rs:1936]: FWIW, I find putting the constraint with the type parameter more easy to read `<P: FsPermissions + 'static>`

File: runtime/ops/fs.rs [hunk 2]
```diff
@@ -36,18 +35,32 @@ use std::rc::Rc;
 use std::time::SystemTime;
 use std::time::UNIX_EPOCH;
 
+pub trait FsPermissions {
+  fn check_read(&mut self, _p: &Path, _api_name: &str) -> Result<(), AnyError>;
```
Comment (dsherret) [inline runtime/ops/fs.rs:39]: Should these not have underscores?

**Review by dsherret — APPROVED** [review]
LGTM

File: runtime/ops/fs.rs (hunk 1 above)
Comment (bartlomieju) [reply runtime/ops/fs.rs:1936]: I don't think that will work, clippy is complaining:
```
error[E0658]: associated type bounds are unstable
   --> runtime/ops/fs.rs:198:17
    |
198 | fn op_open_sync<P: FsPermissions + 'static>(
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #52662 <https://github.com/rust-lang/rust/issues/52662> for more information

```

Comment (dsherret) [reply runtime/ops/fs.rs:1935]: Oh, ok too bad for now.

**Pull request merged**